src/AdminDateFilterCache.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\CoreStandards\AdminDateFilterCache.
+ *
+ * @thanks https://github.com/Automattic/vip-go-mu-plugins-built/blob/master/performance/vip-tweaks.php
+ */
+
+namespace Netzstrategen\CoreStandards;
+
+class AdminDateFilterCache {
+
+  /**
+   * Caches an expensive query for available months/years in media library filters.
+   *
+   * @implements media_library_months_with_files
+   */
+  public static function media_library_months_with_files($months) {
+    // Abort if something else is customizing the value.
+    if (isset($months)) {
+      return $months;
+    }
+    return static::getAvailableMonths('attachment');
+  }
+
+  /**
+   * Caches an expensive query for available months/years in posts filters.
+   *
+   * @implements pre_months_dropdown_query
+   */
+  public static function pre_months_dropdown_query($months, $post_type) {
+    // Abort if something else is customizing the value.
+    if (isset($months)) {
+      return $months;
+    }
+    return static::getAvailableMonths($post_type);
+  }
+
+  /**
+   * Caches results of available months/years for a given post type.
+   *
+   * @param string $post_type
+   *   The post type for which to look up available months/years.
+   *
+   * @return object[]
+   *
+   * @see \WP_List_Table::months_dropdown()
+   * @see wp_enqueue_media()
+   */
+  public static function getAvailableMonths($post_type) {
+    global $wpdb;
+
+    $months = wp_cache_get($post_type, 'admin_date_filter');
```

Comment:
Wouldn't it make sense to prefix the key as they do?  Or do you think the group is enough? 
```suggestion
    $months = wp_cache_get("available_filter_months_$post_type", 'admin_date_filter');
```

Reply:
Not sure, but I think the genetic post type group is fine. We're maintaining the cache item already, so there's no invalidation issue. The only possibility could be that it's then invalidated more often with the whole group, but I think if something does that then it's okay to include this.